src/bots/liquidator.ts

```diff
@@ -600,6 +641,187 @@ export class LiquidatorBot implements Bot {
 			});
 	}
 
+	private getOrderParamsForPerpDerisk(
+		subaccountId: number,
+		position: PerpPosition
+	): OptionalOrderParams | undefined {
+		const nowSec = Math.floor(Date.now() / 1000);
+
+		let baseAssetAmount: BN;
+		if (this.isTwapping()) {
+			const twapProgress = this.twapExecutionProgresses.get(
+				this.getTwapProgressKey(
+					MarketType.PERP,
+					subaccountId,
+					position.marketIndex
+				)
+			);
+			twapProgress.updateProgress(position.baseAssetAmount, nowSec);
+			baseAssetAmount = twapProgress.getExecutionSlice(nowSec);
```

Comment:
is this going to be zero the first time it's hit? since updateProgress will reset the position? maybe need to change `getExecutionSlice` to account for `secElapsed` starting at 0